src/beast/site_scons/site_tools/VSProject.py

```diff
@@ -807,20 +845,10 @@ def getGroup(abspath):
         f.write('</Project>\r\n')
 
     def build(self):
-        try:
-            self.project_file = open(str(self.project_node), 'wb')
-        except IOError, detail:
-            raise SCons.Errors.InternalError('Unable to open "' +
-                str(self.project_node) + '" for writing:' + str(detail))
-        try:
-            self.filters_file = open(str(self.filters_node), 'wb')
-        except IOError, detail:
-            raise SCons.Errors.InternalError('Unable to open "' +
-                str(self.filters_node) + '" for writing:' + str(detail))
-        self.writeProject()
-        self.writeFilters()
-        self.project_file.close()
-        self.filters_file.close()
+        with openNoExe(str(self.project_node)) as project_file:
```

Comment:
I'm really not getting why this couldn't just be `with open(self.project_node, "wb") as project_file:`.